flatland_plugins/src/diff_drive.cpp

```diff
@@ -113,7 +116,7 @@ void DiffDrive::OnInitialize(const YAML::Node& config) {
   // init the values for the messages
   ground_truth_msg_.header.frame_id = odom_frame_id;
   ground_truth_msg_.child_frame_id =
-      tf::resolve("", GetModel()->NameSpaceTF(body_->name_));
+      tf::resolve(ns, GetModel()->NameSpaceTF(body_->name_));
```

Comment:
I'm considering moving flatland from tf to tf2 at some point, which doesn't support prefixes.

This is why I'm calling GetModel()->NameSpaceTF() here. It prefixes all tfs consistently across the model.
https://github.com/avidbots/flatland/blob/master/flatland_server/src/model.cpp#L166

Does your codebase plan on supporting TF2 at some point? The TF maintainers decided that multirobot is better served with multimaster + bridge, which I happen to believe is a giant mess. I was planning on just routing around the problem.

This PR as implemented effectively commits to staying on tf forever and not moving to tf2, which is an option but will likely cause issues with future ros2 support (since tf2 exists for ros2 but tf does not).

Is it possible to use the existing "namespace" parameter when spawning a model via service or defining a model in a world.yaml file? This will result in {namespace: foo, tf: bar} being published with tf name "foo_bar", rather than the tf::resolve output via {ns: foo,  tf: bar} name of "foo/bar". 

I'm hesitant to accept this change as is because it introduces a 2nd parameter that does almost the same thing, with almost the same name ("namespace" on the parent model vrs. "use_namespace" flag on the child plugin).

That said, I'm happy to try to figure out the best way forward to get your simulation goals up and running in Flatland.